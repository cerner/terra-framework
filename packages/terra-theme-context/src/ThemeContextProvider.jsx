import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import ThemeContext from './ThemeContext';

const propTypes = {
  /**
   * The components to be rendered within the context of the ThemeContextProvider. Components rendered here are able to interact with ThemeContextProvider through the ThemeContext.
   */
  children: PropTypes.element.isRequired,
  /**
   * An object containing the name and className of the selected theme.
   */
  theme: PropTypes.shape({
    name: PropTypes.string,
    className: PropTypes.string,
  }),
};

const defaultProps = {
  theme: {},
};

const ThemeContextProvider = ({ theme, children }) => {
  const contextValue = useMemo(() => ({ name: theme.name, className: theme.className }), [theme.name, theme.className]);
  return (
    <ThemeContext.Provider value={contextValue}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeContextProvider.propTypes = propTypes;
ThemeContextProvider.defaultProps = defaultProps;

export default ThemeContextProvider;
