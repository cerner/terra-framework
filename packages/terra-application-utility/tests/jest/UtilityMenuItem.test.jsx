import React from 'react';
import ThemeContextProvider from 'terra-theme-context/lib/ThemeContextProvider';
import HeaderUtilityMenuItem from '../../src/utility/_UtilityMenuItem';

describe('UtilityMenuItem', () => {
  const mockOnChange = jest.fn();
  const title = 'title';
  const content = (<div> test</div>);
  const contentLocation = 'footer';
  const variant = 'header';

  it('should render with default props', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a title', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      title={title}
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with content', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      content={content}
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with a content location', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      contentLocation={contentLocation}
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with isActive', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      hasChevron
      variant={variant}
      isActive
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with isSelected', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      isSelected
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with isSelectable', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      isSelectable
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with hasChevron', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      hasChevron
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with onKeyDown', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      onKeyDown={jest.fn()}
      variant={variant}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with leftInset', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      hasChevron
      variant={variant}
      leftInset
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with rightInset', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      hasChevron
      variant={variant}
      leftInset
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass in a custom prop', () => {
    const wrapper = shallow(<HeaderUtilityMenuItem
      itemKey="key"
      onChange={mockOnChange}
      variant={variant}
      id="test"
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly applies the theme context className', () => {
    const component = mount(
      <ThemeContextProvider theme={{ className: 'orion-fusion-theme' }}>
        <HeaderUtilityMenuItem
          itemKey="key"
          onChange={mockOnChange}
          variant={variant}
        />
      </ThemeContextProvider>,
    );
    expect(component).toMatchSnapshot();
  });
});
