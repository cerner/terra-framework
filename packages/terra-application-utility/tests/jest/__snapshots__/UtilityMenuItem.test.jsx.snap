// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityMenuItem correctly applies the theme context className 1`] = `
<ThemeContextProvider
  theme={
    Object {
      "className": "orion-fusion-theme",
    }
  }
>
  <UtilityMenuItem
    itemKey="key"
    onChange={[MockFunction]}
    title=""
    variant="header"
  >
    <div
      aria-label=""
      className="header-utility-body-item orion-fusion-theme"
      key="key"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <Arrange
        align="center"
        className="default-left-inset default-right-inset"
        fill={
          <div>
            
          </div>
        }
        fillAttributes={
          Object {
            "className": "menu-item-fill",
          }
        }
        fitEnd={null}
        fitStart={null}
      >
        <div
          className="arrange default-left-inset default-right-inset"
        >
          <div
            className="fit center"
          />
          <div
            className="fill center menu-item-fill"
          >
            <div />
          </div>
          <div
            className="fit center"
          />
        </div>
      </Arrange>
    </div>
  </UtilityMenuItem>
</ThemeContextProvider>
`;

exports[`UtilityMenuItem should pass in a custom prop 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  id="test"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with a content location 1`] = `
<Button
  className="header-utility-footer-item"
  isBlock={false}
  isCompact={false}
  isDisabled={false}
  isIconOnly={false}
  isReversed={false}
  onClick={[Function]}
  onKeyDown={[Function]}
  text=""
  type="button"
  variant="neutral"
/>
`;

exports[`UtilityMenuItem should render with a title 1`] = `
<div
  aria-label="title"
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        title
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with content 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
         test
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with default props 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with hasChevron 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with isActive 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with isSelectable 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with isSelected 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with leftInset 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={
      <IconCheckmark
        className="checkmark"
        data-name="Layer 1"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
  />
</div>
`;

exports[`UtilityMenuItem should render with onKeyDown 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-left-inset default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={null}
  />
</div>
`;

exports[`UtilityMenuItem should render with rightInset 1`] = `
<div
  aria-label=""
  className="header-utility-body-item"
  key="key"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <Arrange
    align="center"
    className="default-right-inset"
    fill={
      <div>
        
      </div>
    }
    fillAttributes={
      Object {
        "className": "menu-item-fill",
      }
    }
    fitEnd={null}
    fitStart={
      <IconCheckmark
        className="checkmark"
        data-name="Layer 1"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
  />
</div>
`;
